### PR TITLE
Changelog 143

### PR DESCRIPTION
**Pull request by Undin**

### CODE REVIEW

File: _posts/2021-03-15-changelog-143.markdown [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+## New Features
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6564 -->
+* Provide initial support for [function-like](https://doc.rust-lang.org/reference/procedural-macros.html#function-like-procedural-macros) procedural macros.
+  Now the plugin can expand such procedural macro calls and as a result they
+  automatically get a bunch of some features available for declarative macros:
+  highlighting, name resolution, restricted completion, `Show macro expansion` intentions and others.
+  Note, it's only initial implementation, so it may work in unexpected way in some cases.
+  The feature is disabled for default for now, to turn if on,
+  you should enable `org.rust.cargo.evaluate.build.scripts` and `org.rust.macros.proc` experimental features.
```
Comment (vlad20012) [inline _posts/2021-03-15-changelog-143.markdown:15]: Let's add a note that after enabling these features a user should manually refresh the cargo projects (or restart the IDE)

Comment (Undin) [reply _posts/2021-03-15-changelog-143.markdown:15]: Thanks!
Maybe it's worth just adding the corresponding listeners into `org.rust.cargo.project.model.impl.CargoProjectsServiceImpl`?

File: _posts/2021-03-15-changelog-143.markdown [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  with [new name resolution](https://github.com/intellij-rust/intellij-rust/issues/6217) engine
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6899 -->
+* Exclude more stdlib parts from indexing. Slightly speed up index process
+
+## Fixes
+
+* [New name resolution](https://github.com/intellij-rust/intellij-rust/issues/6217) engine improvements:
+    <!-- https://github.com/intellij-rust/intellij-rust/pull/6872 -->
+    * Find all possible impls for type aliases regardless type aliases number for particular type.
```
Comment (ortem) [inline _posts/2021-03-15-changelog-143.markdown:48]:
```suggestion
    * Find all possible impls for a type alias regardless of the number of aliases defined for the corresponding type.
```
BTW not sure if should we use "impls" or "`impl`s" (there are lots of usages of both in our changelogs)

File: _posts/2021-03-15-changelog-143.markdown [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      Previously, the plugin processed impl blocks for type aliases only
```
Comment (ortem) [inline _posts/2021-03-15-changelog-143.markdown:49]:
```suggestion
      Previously, the plugin processed impl blocks for a type alias only
```

File: _posts/2021-03-15-changelog-143.markdown [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+      when their number for particular type were less than some threshold (10 by default).
```
Comment (ortem) [inline _posts/2021-03-15-changelog-143.markdown:50]:
```suggestion
      when the number of aliases for the corresponding type does not exceed the given threshold (10 by default).
```

File: _posts/2021-03-15-changelog-143.markdown [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+      It should fix name resolution for crates that heavily use type aliases like [nalgebra](https://crates.io/crates/nalgebra) or [winapi](https://crates.io/crates/winapi)
```
Comment (ortem) [inline _posts/2021-03-15-changelog-143.markdown:51]:
```suggestion
      It should fix name resolution for crates that heavily use type aliases, e.g. [nalgebra](https://crates.io/crates/nalgebra) or [winapi](https://crates.io/crates/winapi)
```

File: _posts/2021-03-15-changelog-143.markdown [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+* Fix false-positive [E0252](https://doc.rust-lang.org/error-index.html#E0252) error annotations in some cases
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6891 -->
+* Show proper console output on client when running binary with [Code With Me](https://plugins.jetbrains.com/plugin/14896-code-with-me) plugin (by [@akozlova])
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6876 -->
+* Do not expand invocations of builtin macros into `{}`
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6717 -->
+* Take into account [const generics](https://rust-lang.github.io/rfcs/2000-const-generics.html) by inspections which checks generic parameters/arguments ordering.
```
Comment (ortem) [inline _posts/2021-03-15-changelog-143.markdown:73]:
```suggestion
* Take into account [const generics](https://rust-lang.github.io/rfcs/2000-const-generics.html) in inspections that check generic parameters/arguments ordering.
```

File: _posts/2021-03-15-changelog-143.markdown [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+  It fixes false positive `Type arguments must be declared prior to const arguments` error annotations
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6887 -->
+* Don't throw an exception on changing name of local variable during indexing
```
Comment (ortem) [inline _posts/2021-03-15-changelog-143.markdown:77]:
```suggestion
* Don't throw an exception when changing the name of a local variable during indexing
```

File: _posts/2021-03-15-changelog-143.markdown [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6897 -->
+* Omit unchanged default type parameters of return type by `Extract Function` (`Refactor | Extract Method` or
+  <kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>M</kbd>) refactoring
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6894 -->
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6892 -->
+* Fix false-positive [E0252](https://doc.rust-lang.org/error-index.html#E0252) error annotations in some cases
+
+<!-- https://github.com/intellij-rust/intellij-rust/pull/6891 -->
+* Show proper console output on client when running binary with [Code With Me](https://plugins.jetbrains.com/plugin/14896-code-with-me) plugin (by [@akozlova])
```
Comment (ortem) [inline _posts/2021-03-15-changelog-143.markdown:67]:
```suggestion
* Show proper console output on the client side when running a binary with [Code With Me](https://plugins.jetbrains.com/plugin/14896-code-with-me) plugin (by [@akozlova])
```

File: _posts/2021-03-15-changelog-143.markdown (hunk 2 above)
Comment (Undin) [reply _posts/2021-03-15-changelog-143.markdown:48]: I think `impl` block is more correct

**Pull request merged**